allennlp/modules/seq2seq_decoders/simple_seq_decoder.py

```diff
@@ -0,0 +1,328 @@
+from typing import Dict, List, Tuple
+
+import numpy
+import torch
+import torch.nn.functional as F
+
+from allennlp.data import Vocabulary
+from allennlp.modules.seq2seq_decoders.decoder_cell import DecoderCell
+from allennlp.modules.seq2seq_decoders.seq_decoder import SeqDecoder
+from allennlp.nn import util
+from allennlp.training.metrics import Metric
+
+
+class SimpleSeqDecoder(SeqDecoder):
```

Comment:
What other implementations of the decoder are you imagining?

Reply:
@epwalsh proposed CopyNet